source/docs/software/basic-programming/joystick.rst

```diff
@@ -0,0 +1,59 @@
+.. include:: <isonum.txt>
+
+Joysticks
+=========
+
+FRC\ |reg| joysticks come in all shapes and sizes and virtually any "controller" that can be recognized by Windows can be used as a joystick.  Joysticks are accessed using the ``GenericHID`` class.  This class has two relevant subclasses for preconfigured joysticks but if not listed you could implement your own.  The first is ``Joystick`` which is useful for standar flight joysticks.  The second is ``XboxController`` which work for the Xbox 360, Xbox One, or Logitech F310 (in XInput mode).
+
+The command based way to use the these classes is detailed in the section: :ref:`docs/software/commandbased/binding-commands-to-triggers:Binding Commands to Triggers`
+
+Driver Station Joysticks
+------------------------
+
+A joystick can be used with the Driver Station program to control the robot.  The startup routine will read whatever position the joysticks are in as the center position, therefore, when the station is turned on the joysticks must be at their center position. In general the Driver Station software will try to preserve the ordering of devices between runs but it is a good idea to note what order your devices should be in and check each time you start the Driver Station software that they are correct.  The :ref:`USB Devices Tab <docs/software/driverstation/driver-station:USB Devices Tab>` of the Driver Station is used to setup and configure the joystick for use with the robot.  Pressing a button on a joystick will cause its entry in the table to light up blue and have asterisks appear after the name. To reorder the joysticks simply click and drag.
```

Comment:
I think it's worth noting that it's when the computer is turned on or joysticks are plugged in.